### PR TITLE
Fork the deployment section for dpl v2

### PR DESCRIPTION
**Pull request by svenfuchs**
This follows a similar strategy as with conditions v1 and v2, i.e. we have two pages that link to each other in a "note" blurb, and depending on the status of the release process tweak the test of it. 

Eventually the new page would become the default page, and we'd keep the old one around for as long as we still allow people to use `dpl` v1.

### CODE REVIEW

File: user/deployment-v2.md [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+
+{% include deployments.html %}
+
+To deploy to a custom or unsupported provider, use the [after-success build
+stage](/user/deployment/custom/) or [script provider](/user/deployment/script).
+
+## Maturity Levels
+
+In order to communicate the current development status and maturity of dpl's
+support for a particular service the respective provider is marked with one of
```
Comment (Pezi777) [inline user/deployment-v2.md]:
```suggestion
support for a particular service, the respective provider is marked with one of
```

File: user/deployment-v2.md [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+the following maturity levels, according to the given criteria:
+
+* `dev` - the provider is in development (initial level)
+* `alpha` - the provider is fully tested
+* `beta` - the provider has been in alpha for at least a month, and successful real-world production deployments have been observed
```
Comment (Pezi777) [inline user/deployment-v2.md]:
```suggestion
* `beta` - the provider has been in alpha for at least a month and successful real-world production deployments have been observed
```

File: user/deployment-v2.md [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+* `stable` - the provider has been in beta for at least two months, and there are no open issues that qualify as critical (such as deployments failing, documented functionality broken, etc)
```
Comment (Pezi777) [inline user/deployment-v2.md]:
```suggestion
* `stable` - the provider has been in beta for at least two months and there are no open issues that qualify as critical (such as deployments failing, documented functionality broken, etc.)
```

File: user/deployment-v2.md [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+
+Dpl v2 represents a major rewrite, so support for all providers has been
+reset to `dev` or `alpha`, depending on the test status.
+
+For all levels except `stable` a message will be printed to your build log
+that informs you about the current status.
+
+## Cleaning up the Git working directory
```
Comment (Pezi777) [inline user/deployment-v2.md]:
```suggestion
## Cleaning up the Git Working Directory
```

File: user/deployment-v2.md [hunk 5]
```diff
@@ -0,0 +1,207 @@
...
+
+The previous version of dpl, our deployment integration tooling, used to
```
Comment (Pezi777) [inline user/deployment-v2.md]:
```suggestion
The previous version of dpl - our deployment integration tooling - used to
```

File: user/deployment-v2.md [hunk 6]
```diff
@@ -0,0 +1,207 @@
...
+reset your working directory, and delete all changes made during the build
```
Comment (Pezi777) [inline user/deployment-v2.md]:
```suggestion
reset your working directory and delete all changes made during the build
```

File: user/deployment-v2.md [hunk 7]
```diff
@@ -0,0 +1,207 @@
...
+using `git stash --all`. In order to keep changes one had to opt out using
+`skip_cleanup: true`. This default turned out to be useful only for very few
+providers, and has been changed in dpl v2.
```
Comment (Pezi777) [inline user/deployment-v2.md]:
```suggestion
providers and has been changed in dpl v2.
```

File: user/deployment-v2.md [hunk 8]
```diff
@@ -0,0 +1,207 @@
...
+   ```yaml
+   deploy:
+     provider: s3
+     on:
+       repo: travis-ci/dpl
+   ```
+   {: data-file=".travis.yml"}
+
+* `branch`: name of the branch.
+   If omitted, this defaults to the `app`-specific branch, or `master`. If the branch name is not known ahead of time, you can specify
```
Comment (Pezi777) [inline user/deployment-v2.md]:
```suggestion
   If omitted, this defaults to the `app`-specific branch or `master`. If the branch name is not known ahead of time, you can specify
```

File: user/deployment-v2.md [hunk 9]
```diff
@@ -0,0 +1,207 @@
...
+   `all_branches: true` *instead of* `branch: ` and use other conditions to control your deployment.
+
+* `jdk`, `node`, `perl`, `php`, `python`, `ruby`, `scala`, `go`: for    language runtimes that support multiple versions,
```
Comment (Pezi777) [inline user/deployment-v2.md]:
```suggestion
* `jdk`, `node`, `perl`, `php`, `python`, `ruby`, `scala`, `go`: for language runtimes that support multiple versions,
```

File: user/deployment-v2.md [hunk 10]
```diff
@@ -0,0 +1,207 @@
...
+   you can limit the deployment to happen only on the job that matches a specific version.
+
+* `condition`: deploy when *a single* bash condition evaluates to `true`. This must be a string value, and is equivalent to `if [[ <condition> ]]; then <deploy>; fi`. For example, `$CC = gcc`.
```
Comment (Pezi777) [inline user/deployment-v2.md]:
```suggestion
* `condition`: deploy when *a single* bash condition evaluates to `true`. This must be a string value and is equivalent to `if [[ <condition> ]]; then <deploy>; fi`. For example, `$CC = gcc`.
```

File: user/deployment-v2.md [hunk 11]
```diff
@@ -0,0 +1,207 @@
...
+    tags: true
+    rvm: 2.0.0
+```
+{: data-file=".travis.yml"}
+
+### Adding a deployment provider
+
+We are working on adding support for other PaaS providers. If you host your application with a provider not listed here and you would like to have Travis CI automatically deploy your application, please [get in touch](mailto:support@travis-ci.com).
+
+If you contribute to or experiment with the [deploy tool](https://github.com/travis-ci/dpl) make sure you use the edge version from GitHub:
```
Comment (Pezi777) [inline user/deployment-v2.md]:
```suggestion
If you contribute to or experiment with the [deploy tool](https://github.com/travis-ci/dpl), make sure you use the edge version from GitHub:
```

File: user/deployment-v2.md [hunk 12]
```diff
@@ -0,0 +1,207 @@
...
+```yaml
+deploy:
+  provider: awesome-experimental-provider
+  edge: true
+```
+{: data-file=".travis.yml"}
+
+## Pull Requests
+
+Note that pull request builds skip the deployment step altogether.
```
Comment (Pezi777) [inline user/deployment-v2.md]:
```suggestion
> Note that pull request builds skip the deployment step altogether.
```

File: user/deployment-v2.md [hunk 13]
```diff
@@ -0,0 +1,207 @@
+---
+title: Deployment (v2)
+layout: en
+---
+
+> This page documents deployments using the next major version dpl v2 which
+> currently is in a preview release phase. Please see our blog post on details
```
Comment (MariadeAnton) [inline user/deployment-v2.md]:
```suggestion
> currently is in a preview release phase. Please see [the announcement blog post](https://blog.travis-ci.com/2019-08-27-deployment-tooling-dpl-v2-preview-release) on details
```

File: user/deployment-v2.md [hunk 14]
```diff
@@ -0,0 +1,207 @@
+---
+title: Deployment (v2)
+layout: en
+---
+
+> This page documents deployments using the next major version dpl v2 which
```
Comment (MariadeAnton) [inline user/deployment-v2.md]:
```suggestion
> ALPHA This page documents deployments using the next major version dpl v2 which
```

File: user/deployment-v2.md [hunk 15]
```diff
@@ -0,0 +1,207 @@
...
+> currently is in a preview release phase. Please see our blog post on details
+> about the release process. Documentation for dpl v1, the current default
+> version, can be found [here](/user/deployments).
```
Comment (MariadeAnton) [inline user/deployment-v2.md]:
```suggestion
> version, can be found [here](/user/deployments). {: alpha}
```

File: user/deployment-v2.md [hunk 16]
```diff
@@ -0,0 +1,207 @@
...
+
+## Supported Providers
+
+Continuous Deployment to the following providers is supported:
+
+{% include deployments.html %}
+
+To deploy to a custom or unsupported provider, use the [after-success build
+stage](/user/deployment/custom/) or [script provider](/user/deployment/script).
```
Comment (MariadeAnton) [inline user/deployment-v2.md]:
```suggestion
step](/user/deployment/custom/) or [script provider](/user/deployment/script).
```
consistency wise, I believe stages are the ones people can define through build stages while the different sections that run during the build life-cycle are steps

File: user/deployment-v2.md [hunk 17]
```diff
@@ -0,0 +1,207 @@
...
+    branch: release
+    condition: $MY_ENV = super_awesome
+```
+{: data-file=".travis.yml"}
+
+When *all* conditions specified in the `on:` section are met, your build will deploy.
+
+Use the following options to configure conditional deployment:
+
+* `repo`: in the form `owner_name/repo_name`. Deploy only when the build occurs on a particular repository. For example
```
Comment (MariadeAnton) [inline user/deployment-v2.md]:
```suggestion
* `repo`: in the form `owner_name/repo_name`. Deploy only when the build occurs on a particular repository. For example:
```

File: user/deployment-v2.md [hunk 18]
```diff
@@ -0,0 +1,207 @@
...
+   ```yaml
+   deploy:
+     provider: s3
+     on:
+       repo: travis-ci/dpl
+   ```
+   {: data-file=".travis.yml"}
+
+* `branch`: name of the branch.
+   If omitted, this defaults to the `app`-specific branch or `master`. If the branch name is not known ahead of time, you can specify
```
Comment (MariadeAnton) [inline user/deployment-v2.md]:
```suggestion
   If omitted, this defaults to the `app`-specific branch or `master`. If the branch name is not known ahead of time, you can specify
```

what does the `app`-specific branch mean? The branch that the specific deployment provider would look for by default?

File: user/deployment-v2.md [hunk 19]
```diff
@@ -0,0 +1,207 @@
...
+  skip_cleanup: true
+  on:
+    tags: true
+    rvm: 2.0.0
+```
+{: data-file=".travis.yml"}
+
+### Adding a deployment provider
+
+We are working on adding support for other PaaS providers. If you host your application with a provider not listed here and you would like to have Travis CI automatically deploy your application, please [get in touch](mailto:support@travis-ci.com).
```
Comment (MariadeAnton) [inline user/deployment-v2.md]:
```suggestion
We are working on adding support for other PaaS providers. If you host your application with a provider not listed here and you would like to have Travis CI automatically deploy your application, please [get in touch](mailto:support@travis-ci.com?subject:New%20deployment%20provider%20proposal).
```

File: _layouts/en.html [hunk 20]
```diff
@@ -18,6 +18,27 @@
             <h1 class="title">{{ page.title }}</h1>
           {% endunless %}
 
+          {% if page.deploy == 'v1' %}
+            <blockquote>
+              <p>
+                This page documents deployments using dpl v1 which currently is the default
+                version. The next major version dpl v2 will be released soon, and we recommend
+                starting to use it. Please see <a href="https://blog.travis-ci.com/2019-08-27-deployment-tooling-dpl-v2-preview-release">our blog post</a>
+                for details. Documentation for dpl v2 can be found <a href="/user/deployments">here</a>.
```
Comment (MariadeAnton) [inline _layouts/en.html]:
```suggestion
                for details. <a href="/user/deployment">dpl v2 documentation can be found here</a>.
```

I believe in linking longer despcriptions than here, this helps with screen readers and building trust :) 

This also fixes the link from `/deployments` to `/deployment`

File: _layouts/en.html [hunk 21]
```diff
@@ -18,6 +18,27 @@
...
+              </p>
+            </blockquote>
+          {% elsif page.deploy == 'v2' %}
+            <blockquote>
+              <p>
+                This page documents deployments using the next major version dpl v2 which
+                currently is in a preview release phase. Please see <a href="https://blog.travis-ci.com/2019-08-27-deployment-tooling-dpl-v2-preview-release">our blog post</a>
+                for details. Documentation for dpl v1, the current default version, can be
+                found <a href="/user/deployments">here</a>.
```
Comment (MariadeAnton) [inline _layouts/en.html]:
```suggestion
                Check <a href="/user/deployment">dpl v1 documentation here</a>.
```

File: _layouts/en.html [hunk 22]
```diff
@@ -18,6 +18,27 @@
...
+                starting to use it. Please see <a href="https://blog.travis-ci.com/2019-08-27-deployment-tooling-dpl-v2-preview-release">our blog post</a>
+                for details. Documentation for dpl v2 can be found <a href="/user/deployments">here</a>.
+              </p>
+            </blockquote>
+          {% elsif page.deploy == 'v2' %}
+            <blockquote>
+              <p>
+                This page documents deployments using the next major version dpl v2 which
+                currently is in a preview release phase. Please see <a href="https://blog.travis-ci.com/2019-08-27-deployment-tooling-dpl-v2-preview-release">our blog post</a>
+                for details. Documentation for dpl v1, the current default version, can be
```
Comment (MariadeAnton) [inline _layouts/en.html]:
```suggestion
                for details. The current default version is dpl v1. 
```

File: user/deployment-v2/azure-web-apps.md [hunk 23]
```diff
@@ -0,0 +1,99 @@
...
+As this deployment strategy relies on `git`, be mindful that the deployment will
+honor `.gitignore`.
+
+If your `.gitignore` file matches something that your build creates, use
+[`before_deploy`](#running-commands-before-and-after-deploy) to change
+its content.
+
+### Running commands before and after deploy
+
+Sometimes you want to run commands before or after deploying. You can use the `before_deploy` and `after_deploy` stages for this. These will only be triggered if Travis CI is actually deploying.
```
Comment (MariadeAnton) [inline user/deployment-v2/azure-web-apps.md]:
```suggestion
Sometimes you want to run commands before or after deploying. You can use the `before_deploy` and `after_deploy` build steps for this. These will only be triggered if Travis CI is actually deploying.
```

File: user/deployment-v2/bintray.md [hunk 24]
```diff
@@ -0,0 +1,200 @@
...
+Builds triggered from Pull Requests will never trigger a deploy.
+
+### Conditional Deploys
+
+You can deploy only when certain conditions are met.
+See [Conditional Releases with `on:`](/user/deployment#conditional-releases-with-on).
+
+### Running commands before and after deploy
+
+Sometimes you want to run commands before or after deploying. You can use the `before_deploy` and `after_deploy` stages for this. These will only be triggered if Travis CI is actually deploying.
```
Comment (MariadeAnton) [inline user/deployment-v2/bintray.md]:
```suggestion
Sometimes you want to run commands before or after deploying. You can use the `before_deploy` and `after_deploy` build steps for this. These will only be triggered if Travis CI is actually deploying.
```

File: user/deployment-v2/bitballoon.md [hunk 25]
```diff
@@ -0,0 +1,54 @@
...
+  access-token:
+    secure: "YOUR ENCRYPTED ACCESS TOKEN"
+  local-dir: "_build/"
+```
+{: data-file=".travis.yml"}
+
+## Running commands before and after deploy
+
+Sometimes you want to run commands before or after deploying. You can use
+the `before_deploy` and `after_deploy` stages for this. These will only be
```
Comment (MariadeAnton) [inline user/deployment-v2/bitballoon.md]:
```suggestion
the `before_deploy` and `after_deploy` build steps for this. These will only be
```

File: user/deployment-v2.md (hunk 18 above)
Comment (svenfuchs) [reply user/deployment-v2.md]: I don't know ... I think I've just copied that text from the existing page?

**Pull request merged**